File: TOOLS/lua/pause-when-minimize.lua
-- This script pauses playback when minimizing the window, and resumes playback
-- if it's brought back again. If the player was already paused when minimizing,
-- then try not to mess with the pause state.

local did_minimize = false

mp.observe_property("window-minimized", "bool", function(_, value)
    local pause = mp.get_property_native("pause")
    if value == true then
        if pause == false then
            mp.set_property_native("pause", true)
            did_minimize = true
        end
    elseif value == false then
        if did_minimize and (pause == true) then
            mp.set_property_native("pause", false)
        end
        did_minimize = false
    end
end)

File: TOOLS/python/pause-when-minimize.py
# This script pauses playback when minimizing the window, and resumes playback
# if it's brought back again. If the player was already paused when minimizing,
# then try not to mess with the pause state.
from mpvclient import mpv  # type: ignore

did_pause_at_minimize = False

@mpv.observe_property("window-minimized", mpv.MPV_FORMAT_NODE)
def on_window_minimized(value):
    pause = mpv.get_property_bool("pause")
    global did_pause_at_minimize

    if value:
        if not pause:
            mpv.set_property_bool("pause", True)
            did_pause_at_minimize = True
    else:
        if did_pause_at_minimize and pause:
            mpv.set_property_bool("pause", False)
            did_pause_at_minimize = False  # Reset to False for probable next cycle
